### PR TITLE
Merge record transformation

### PR DESCRIPTION
**Pull request by Anumodana**
This can be used to merge an array of hash that contains the specified keys into the same hash.

### Input
    # params: { join_keys: [ :Platform, :Level ] }

    [{ Platform: 'android', Level: 1, Percent: 0.0 },
     { Platform: 'ios', Level: 1, Percent: 19.2 },
     { Platform: 'android', Level: 2, Percent: 4.2 },
     { Platform: 'android', Level: 3, Percent: 2.9 },
     { Platform: 'android', Level: 1, MinScore: 2034 },
     { Platform: 'ios', Level: 1, MinScore: 1000 },
     { Platform: 'android', Level: 1, HighScore: 55555 },
    { Platform: 'ios', Level: 1, Fail: 2 }]

### Output
    [{ Platform: 'android', Level: 1, Percent: 0.0, MinScore: 2034, HighScore: 55555 },
     { Platform: 'android', Level: 2, Percent: 4.2 },
     { Platform: 'android', Level: 3, Percent: 2.9 },
     { Platform: 'ios', Level: 1, Percent: 19.2, MinScore: 1000, Fail: 2 }]

### CODE REVIEW

File: spec/optimus_prime/destinations/merge_record_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        next: ['dest_d']
+      },
+      dest_d: { class: 'OptimusPrime::Destinations::MyDestination' }
+    }.merge(sources))
+  end
+
+  it 'should merge a record to an array of hash' do
+    pipeline.start
+    expect(pipeline.started?).to be true
+    expect(pipeline.finished?).to be false
```
Comment (omarkhan) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: You don't need to test these here, we are already testing them in the pipeline test

File: spec/optimus_prime/destinations/merge_record_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    pipeline.wait
+    expect(pipeline.finished?).to be true
+    @results = pipeline.steps.values_at(:dest_d)[0].written
```
Comment (omarkhan) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: No need for an instance variable here

Comment (omarkhan) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: No need for `values_at` either, just call `pipeline.steps[:dest_d]`

**Comment by omarkhan** [comment]
Looks good :smile:

File: spec/optimus_prime/destinations/merge_record_spec.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+      def each
+        @events.each { |event| yield event }
+      end
+    end
+  end
+
+  module Destinations
+    class MyDestination < Destination
+      attr_reader :written
```
Comment (Anumodana) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: @omarkhan I have a question. When I run `pronto run` locally, I got the following error,

    spec/optimus_prime/destinations/merge_record_spec.rb:21 I: Use `attr_writer` to define trivial writer methods.

Should I omit this alert or try to fix it?

Comment (omarkhan) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: Should `@written` only contain the last record? Shouldn't it be an array?

Comment (omarkhan) [inline spec/optimus_prime/destinations/merge_record_spec.rb]: You can ignore this alert, it doesn't really matter :smile:

**Comment by Anumodana** [comment]
@omarkhan Fixed :smiley:  I'll merge it now.

**Pull request merged**